File: docs/examples/add-to-your-application/node/oso.js
const { Oso } = require('oso');
const { Repository, User } = require('./models');

async function initOso() {
  // Initialize the Oso object. This object is usually
  // used globally throughout an application.
  const oso = new Oso();

  // Tell Oso about the data that you will authorize.
  // These types can be referenced in the policy.
  oso.registerClass(User);
  oso.registerClass(Repository);

  // Load your policy file.
  await oso.loadFiles(['main.polar']);

  return oso;
}

module.exports = {
  initOso: initOso
};

File: docs/examples/add-to-your-application/python/app/oso.py
from pathlib import Path

from oso import Oso

from .models import User, Repository

# Initialize the Oso object. This object is usually
# used globally throughout an application.
oso = Oso()

# Tell Oso about the data that you will authorize.
# These types can be referenced in the policy.
oso.register_class(User)
oso.register_class(Repository)

# Load your policy file.
oso.load_files([Path(__file__).parent / "main.polar"])
